Library/include/CSP/Systems/Assets/AssetSystem.h

```diff
@@ -244,8 +244,12 @@ class CSP_API AssetSystem : public SystemBase
     /// @brief Creates a new material backed by an AssetCollection/Asset.
     /// @param Name const csp::common::String& : The name of the new material.
     /// @param SpaceId const csp::common::String& : The space id this material is associated with.
+    /// @param Metadata csp::common::Map<csp::common::String, csp::common::String>& : The metadata to be associated with the created asset colection.
+    /// @param AssetTags csp::common::Array<csp::common::String>& : Tags to be associated with the created asset collection.
     /// @param Callback GLTFMaterialResultCallback : Callback when asynchronous task finishes.
-    CSP_ASYNC_RESULT void CreateMaterial(const csp::common::String& Name, const csp::common::String& SpaceId, GLTFMaterialResultCallback Callback);
+    CSP_ASYNC_RESULT void CreateMaterial(const csp::common::String& Name, const csp::common::String& SpaceId,
+        csp::common::Map<csp::common::String, csp::common::String> Metadata, const csp::common::Array<csp::common::String> AssetTags,
+        GLTFMaterialResultCallback Callback);
```

Comment:
[fix] In all interfaces, you should be using const-ref to prevent copying.

```suggestion
    CSP_ASYNC_RESULT void CreateMaterial(const csp::common::String& Name, const csp::common::String& SpaceId,
        const csp::common::Map<csp::common::String, csp::common::String>& Metadata, const csp::
        common::Array<csp::common::String>& AssetTags,
        GLTFMaterialResultCallback Callback);
```

Reply:
Rapid C++ lesson. What does adding the & do?

Reply:
It means that rather than copying the value, the value will be passed by reference, (ie just a pointer)
You won't need to update any other code other than the function declaration and definition, since the semantics of a value param (no &) and a reference param are identical.

Reply:
You'll note that in this function, the parameters are passed down into `CreateAssetCollection` and then later into `CreatePrototypeDto`, both of which take `const &` parameters. This way, a copy is never made and we use the same memory the entire chain.

Reply:
(It also allows our csharp generators to work, since they only account for const-ref params for complex types ... apparently)

Reply:
Thanks